wolfssl/src/lib.rs

```diff
@@ -18,7 +18,7 @@ pub use context::*;
 pub use rng::*;
 pub use ssl::*;
 
-pub use error::{Error, Poll, Result};
+pub use error::{Error, FatalError, FatalErrorKind, Poll, Result};
```

Comment:
I wonder if `FatalErrorKind` should just be called `ErrorKind` and if we actually need both `FatalError`[^1] and `FatalErrorKind`?


WDYT of just replacing `FatalError` with `ErrorKind`:

```rust
pub enum ErrorKind {
    /// Domain name mismatch error
    DomainNameMismatch,
    /// All other wolfssl fatal errors
    Other{ what: String, code: c_int},
}
```
(note `Other` here now has the contents which `FatalError` has today).

Then Error is
```
pub enum Error {
    /// During secure renegotiation, if application data is found, we must call
    /// `wolfssl_read` to extract the data. If that `wolfssl_read` call fails,
    /// this error will be generated.
    #[error("App Data: {0}")]
    AppData(ErrorKInd),
    /// Top-level errors from WolfSSL API invocations.
    #[error("Fatal: {0}")]
    Fatal(ErrorKind),
}

impl Error {
   fn kind(&self) -> ErrorKind { ... }
}
```


[^1]: I think `FatalError` is actually a bad name, conceptually it's just the specific info about an error, it's no more or less fatal than any other error. If it weren't for the AppData vs other error use case it'd just be called `Error`. But that's a preexsiting issue arising from the whole AppData thing.